### PR TITLE
Add cmsswdata g4data numpy pytools

### PR DESCRIPTION
**Pull request by mrodozov**
this changes are needed for g4data, cmsswdata, pcm_utils, systemtools, coral, python, python3 and fastjet

### CODE REVIEW

**Comment by smuzaffar** [comment]
please fix https://github.com/cms-sw/scram-tools/blob/main/tools/boost/env.sh#L7 and make sure that it return non-empty value. looks like for `iostreams` ( https://github.com/cms-sw/scram-tools/blob/main/tools/boost/env.sh#L20 ) it is returnign emoptry string. We should make sure that `getLibName` fail if it does not find the lib. Thsi will allow us to catch the error at build time

File: tools/fastjet/fastjet.xml [hunk 1]
```diff
@@ -11,6 +11,6 @@
       <environment name="INCLUDE" default="$FASTJET_BASE/include"/>
     </client>
     <runtime name="ROOT_INCLUDE_PATH" value="$INCLUDE" type="path"/>
-    <runtime name="PYTHON27PATH" value="$FASTJET_BASE/lib/python@PYTHONV@/site-packages" type="path"/>
+    <runtime name="PYTHON27PATH" value="$FASTJET_BASE/lib/python@PYTHON3V@/site-packages" type="path"/>
```
Comment (smuzaffar) [inline tools/fastjet/fastjet.xml:14]: we do not need this line. remove it and remove then fastjet/env.sh

**Comment by smuzaffar** [comment]
@mrodozov , I do not see equivalent of https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_12_0_X/master/gcc-toolfile.spec#L208-L216 in gcc/env.sh

**Comment by mrodozov** [comment]
> @mrodozov , I do not see equivalent of https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_12_0_X/master/gcc-toolfile.spec#L208-L216 in gcc/env.sh

this one is in scramv1-tool-conf 
https://github.com/cms-sw/cmsdist/pull/7110/files#diff-f577b1d48bee68be436d53745ccd519b215ae203e46f4661c093eb977a06d717R45-R55

but if CMSDIST_DIR is always exported and a variable for %ppc64le_build_flags is added instead of 
COMP_ARCH_SPECIFIC_FLAGS (which can go away) I can move it in the gcc/env.sh

**Comment by smuzaffar** [comment]
ah right, I think we should always set CMSDIST_DIR but `%{ppc64le_build_flags}` is bit tricky. I think what you have in https://github.com/cms-sw/cmsdist/pull/7110/files#diff-f577b1d48bee68be436d53745ccd519b215ae203e46f4661c093eb977a06d717R45-R55 looks good. Just move the https://github.com/cms-sw/cmsdist/pull/7110/files#diff-f577b1d48bee68be436d53745ccd519b215ae203e46f4661c093eb977a06d717R42 line out of condition ( so that CMSDIST is always set.

**Comment by mrodozov** [comment]
hmm when the *tool-conf files are added the compilation_flags include it works the only case is how to add the flags to gcc if it's built outside cmssw-tool-conf coral-tool-conf or fwlite-tool-conf.
I can check if the COMP_ARCH_SPECIFIC_FLAGS is set and if it isn't - parse the $CMSDIST_DIR/compilation_flags.file inside gcc/env.sh

File: tools/gcc/env.sh [hunk 2]
```diff
@@ -25,6 +25,12 @@ export COMPILER_VERSION_MINOR=$(echo $COMPILER_VERSION | cut -d'.' -f2)
 # *** USE @VARIABLE@ plus associated environment variable to customize. ***
 # DO NOT DUPLICATE the toolfile template.
 
+# if the architecture is ppc64le, search the CMSDIST_DIR for ppc64 compilation flags
+PPC_COMP_FLAGS=""
+if [[ $(arch) == ppc64le ]] ; then
+PPC_COMP_FLAGS=`cat ${CMSDIST_DIR}/compilation_flags.file | grep "define ppc64le_build_flags" | cut -d" " -f3-`
+fi
```
Comment (smuzaffar) [inline tools/gcc/env.sh:32]: I think the logic you have previously was much better i.e. setting COMP_ARCH_SPECIFIC_FLAGS from scramv1-tool-conf.file

Comment (smuzaffar) [reply tools/gcc/env.sh:32]: was that not working?

Comment (mrodozov) [reply tools/gcc/env.sh:32]: the prev logic required including compilation_flags.file in every *tool-conf, which also works

Comment (smuzaffar) [reply tools/gcc/env.sh:32]: you can jsut `##INCLUDE compilation_flags` in scramv1-tool-conf.file which is included in all tool-confs

Comment (smuzaffar) [reply tools/gcc/env.sh:32]: @mrodozov , recursive `## INCLUDE ` should work as it is working for tensorflow-sources
https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_12_0_X/master/tensorflow-sources.spec#L7 
```
## INCLUDE tensorflow-sources
```
https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_12_0_X/master/tensorflow-sources.file#L14-L15
```
## INCLUDE tensorflow-requires
## INCLUDE compilation_flags
```

Comment (smuzaffar) [reply tools/gcc/env.sh:32]: note that there is space after `##`

Comment (mrodozov) [reply tools/gcc/env.sh:32]: there is one difference
tf.spec uses ## INCLUDE tf.file which uses include compilation_flags
cmssw-tool-conf.spec uses ## IMPORT scramv1-tool-conf which uses include compilation_flags

Comment (smuzaffar) [reply tools/gcc/env.sh:32]: ah ok, replace `## IMPORT scramv1-tool-conf ` with `## INCLUDE scramv1-tool-conf ` in all tool-confs

**Pull request merged**